fava/application.py

```diff
@@ -325,6 +326,17 @@ def api_add_document():
             return api_error('{} already exists.'.format(filepath))
 
         file.save(filepath)
+
+        if request.form.get('bfilename', None):
+            # TODO use next_statement_key to calculate the metadata key to
+            # support multiple statement keys
```

Comment:
I wrote a helper-function in `fava.util.__init__` to get the next key from an array of metadata-keys, so if `statement:` is already present it will return `statement-2` and so forth. 

At this point in the code we only know which filename and lineno this metadata key should go to, but nothing about the entry itself, eg. the existing metadata-keys for it. 

@yagebu Do you have an idea how we could get an array of the metadata-keys for that entry at this point in the code?